src/Stratis.Bitcoin.Features.PoA/BasePoAFeatureConsensusRules/PoAHeaderSignatureRule.cs

```diff
@@ -39,77 +34,55 @@ public override void Initialize()
+                if (mightBeInsufficient)
                 {
-                    // Gather all past and present mining public keys.
-                    IEnumerable<PubKey> genesisFederation = ((PoAConsensusOptions)this.network.Consensus.Options).GenesisFederationMembers.Select(m => m.PubKey);
-                    var knownKeys = new HashSet<PubKey>(genesisFederation);
-                    foreach (Poll poll in this.votingManager.GetApprovedPolls().Where(x => ((x.VotingData.Key == VoteKey.AddFederationMember) || (x.VotingData.Key == VoteKey.KickFederationMember))))
-                    {
-                        IFederationMember federationMember = ((PoAConsensusFactory)(this.network.Consensus.ConsensusFactory)).DeserializeFederationMember(poll.VotingData.Data);
-                        knownKeys.Add(federationMember.PubKey);
-                    }
-
-                    // Try to provide the public key that signed the block.
-                    var signature = ECDSASignature.FromDER(header.BlockSignature.Signature);
-                    for (int recId = 0; recId < 4; recId++)
-                    {
-                        PubKey pubKeyForSig = PubKey.RecoverFromSignature(recId, signature, header.GetHash(), true);
-                        if (pubKeyForSig == null)
-                        {
-                            this.Logger.LogDebug($"Could not match candidate keys to any known key.");
-                            break;
-                        }
-
-                        this.Logger.LogDebug($"Attempting to match candidate key '{ pubKeyForSig.ToHex() }' to known keys.");
-
-                        if (!knownKeys.Any(pk => pk == pubKeyForSig))
-                            continue;
-
-                        IEnumerable<PubKey> modifiedFederation = this.votingManager?.GetModifiedFederation(context.ValidationContext.ChainedHeaderToValidate).Select(m => m.PubKey) ?? genesisFederation;
-
-                        this.Logger.LogDebug($"Block {context.ValidationContext.ChainedHeaderToValidate}:{context.ValidationContext.ChainedHeaderToValidate.Header.Time} is signed by '{pubKeyForSig.ToHex()}' but expected '{pubKey}' from: { string.Join(" ", modifiedFederation.Select(pk => pk.ToHex()))}.");
-
-                        break;
-                    };
+                    // Mark header as insufficient to avoid banning the peer that presented it.
+                    // When we advance consensus we will be able to validate it.
+                    context.ValidationContext.InsufficientHeaderInformation = true;
                 }
-                catch (Exception) { }
 
                 this.Logger.LogDebug("(-)[INVALID_SIGNATURE]");
                 PoAConsensusErrors.InvalidHeaderSignature.Throw();
             }
+
+            uint roundTime = this.slotsManager.GetRoundLengthSeconds(federation.Count);
+
+            ChainedHeader prevHeader = context.ValidationContext.ChainedHeaderToValidate.Previous;
```

Comment:
Maybe just add a comment here explaining that we are checking the last round if blocks so that its clear what the intention is.